### PR TITLE
Fix no-return false positives in static analyzer build

### PR DESCRIPTION
**Pull request by janvorli**
There were about 800 false positive issues reported in the clang status analyzer
build caused by the fact that various forms of asserts were not considered
by the analyzer as not returning.

This change adds __attribute__((analyzer_noreturn)) (wrapped in a macro) to
those assertion functions.

### CODE REVIEW

**Comment by janvorli** [comment]
@dotnet-bot test Ubuntu arm Cross Release crossgen_comparison Build and Test please

**Comment by janvorli** [comment]
@dotnet-bot test Windows_NT x86 Release Innerloop Build and Test please

File: src/inc/debugmacros.h [hunk 1]
```diff
@@ -29,6 +30,7 @@ bool GetStackTraceAtContext(SString & s, struct _CONTEXT * pContext);
 void _cdecl DbgWriteEx(LPCTSTR szFmt, ...);
 bool _DbgBreakCheck(LPCSTR szFile, int iLine, LPCSTR szExpr, BOOL fConstrained = FALSE);
 
+ANALYZER_NORETURN
```
Comment (franksinankaya) [inline src/inc/debugmacros.h]: GCC likes extern at the beginning of the line.

Comment (janvorli) [reply src/inc/debugmacros.h]: Ok, let me change that so that you don't have to patch it afterwards.

Comment (franksinankaya) [reply src/inc/debugmacros.h]: thanks a lot

**Comment by janvorli** [comment]
@dotnet-bot test Windows_NT x86 Release Innerloop Build and Test  please

**Pull request merged**